apps/rush-lib/src/logic/buildCache/AmazonS3BuildCacheProvider.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { Terminal } from '@rushstack/node-core-library';
+
+import { EnvironmentConfiguration, EnvironmentVariableNames } from '../../api/EnvironmentConfiguration';
+import { CloudBuildCacheProviderBase } from './CloudBuildCacheProviderBase';
+import { S3Client, GetObjectCommand, PutObjectCommand, GetObjectCommandOutput } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+import { CredentialCache, ICredentialCacheEntry } from '../CredentialCache';
+import { RushConstants } from '../RushConstants';
+import { defaultProvider as awsCredentialsProvider } from '@aws-sdk/credential-provider-node';
+
+interface IAmazonS3Credentials {
+  accessKeyId: string;
+  secretAccessKey: string;
+}
+
+export interface IAmazonS3BuildCacheProviderOptions {
+  s3Bucket: string;
+  s3Region: string;
+  s3Prefix?: string;
+  isCacheWriteAllowed: boolean;
+}
+
+export class AmazonS3BuildCacheProvider extends CloudBuildCacheProviderBase {
+  private readonly _s3Bucket: string;
+  private readonly _s3Region: string;
+  private readonly _s3Prefix: string | undefined;
+  private readonly _environmentWriteCredential: string | undefined;
+  private readonly _isCacheWriteAllowedByConfiguration: boolean;
+
+  public get isCacheWriteAllowed(): boolean {
+    return this._isCacheWriteAllowedByConfiguration || !!this._environmentWriteCredential;
+  }
+
+  private __s3Client: S3Client | undefined;
+
+  public constructor(options: IAmazonS3BuildCacheProviderOptions) {
+    super();
+    this._s3Bucket = options.s3Bucket;
+    this._s3Region = options.s3Region;
+    this._s3Prefix = options.s3Prefix;
+    this._environmentWriteCredential = EnvironmentConfiguration.buildCacheWriteCredential;
+    this._isCacheWriteAllowedByConfiguration = options.isCacheWriteAllowed;
+  }
+
+  private _deserializeCredentials(credentialString: string | undefined): IAmazonS3Credentials | undefined {
+    if (!credentialString) {
+      return undefined;
+    }
+    const splitIndex: number = credentialString.indexOf(':');
+    if (splitIndex === -1) {
+      return undefined;
```

Comment:
Should this throw an error saying that the credential is in an unexpected format?